src/openslide-vendor-dicom.c

```diff
@@ -233,8 +233,7 @@ static bool dicom_detect(const char *filename,
   }
 
   DcmError *dcm_error = NULL;
-  g_autoptr(DcmDataSet) meta =
-    dcm_filehandle_read_file_meta(&dcm_error, filehandle);
+  const DcmDataSet *meta = dcm_filehandle_get_file_meta(&dcm_error, filehandle);
```

Comment:
Do we still need this call at all?  We don't use the value, and if `_openslide_dicom_open` now does enough to verify that this is a valid DICOM file, we could drop it.

The `G_DEFINE_AUTOPTR_CLEANUP_FUNC` for `DcmDataSet` is now unused and can be dropped.

Reply:
I think we still need it -- `dcm_filehandle_create()` only checks that it can open the file. All the "is this a DICOM?" tests are in `dcm_filehandle_get_file_meta()`.

Perhaps `dcm_filehandle_create()` should do a few checks too?

Reply:
Ah, okay.  I don't have an opinion about where libdicom should do the checks, but as it stands, makes sense to keep the call as is.